api/v1beta1/hetznercluster_webhook.go

```diff
@@ -109,6 +149,55 @@ func (r *HetznerCluster) ValidateCreate() (admission.Warnings, error) {
 		if err := isNetworkZoneSameForAllRegions(r.Spec.ControlPlaneRegions, nil); err != nil {
 			allErrs = append(allErrs, err)
 		}
+	} else {
+		// If ID is given check that all other network settings are empty.
+		if r.Spec.HCloudNetwork.ID != nil {
```

Comment:
what if we do the following: We ignore all other properties if the ID is specified and overwrite it with the correct value from the actual network? Seems a bit unnecessary to me to have all this validation, because in the end "ignoring and overwriting" would be also fine I guess. Then we also don't have to change to pointers

I'm okay with both ways though!

Reply:
@janiskemper 

> what if we do the following: We ignore all other properties if the ID is specified and overwrite it with the correct value from the actual network?

I would like to have a clear separation between Spec and Status. We can write the actual values into the Status, but I think writing them to the Spec can create confusion (and is against the api guidelines of Kubernetes).

Reply:
happens for example with the ProviderID as well @guettli

Reply:
> happens for example with the ProviderID as well @guettli

Yes, I know that there are exceptions to the api guidelines. Nevertheless, the guidelines make sense, and we should break them (except there are good reasons).

Reply:
alright, it happens in multiple places (providerID, ControlPlaneEndpoint, etc.) but I am fine to not do it in this case, because we currently don't do it for the network.